### PR TITLE
add restart condition to yaml, service file

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: systemd/systemd.go [hunk 1]
```diff
@@ -103,6 +104,58 @@ type Systemd interface {
...
+	"always":      RestartAlways,
+}
+
+// ErrUnknownRestartCondition is returned when trying to unmarshal an unknown restart condition
+var ErrUnknownRestartCondition = errors.New("invalid restart condition")
+
+func (rc RestartCondition) String() string {
+	s := string(rc)
+	if s == "" {
+		s = string(RestartOnFailure)
```
Comment (niemeyer) [inline systemd/systemd.go]: This feels a bit like a lie.. the rc is not actually a RestartOnFailure, so its string representation shouldn't claim to be it. Imagine us trying to debug an issue where we look at the rc and it just magically changes somewhere.

The actual truth is that we want GenServiceFile to present a default if the rc is unset, so I'd suggest doing that there.

**Comment by niemeyer** [comment]
LGTM assuming the detail above is addressed.

**Comment by mvo5** [comment]
:+1:

**Comment by elopio** [comment]
@chipaca: Sorry for comming late to the party, but I think this needs a documentation update.

**Comment by chipaca** [comment]
Done that in #331.

**Pull request merged**